Library/Homebrew/brew.rb

```diff
@@ -86,6 +86,10 @@ class MissingEnvironmentVariables < RuntimeError; end
   require "commands"
   require "settings"
 
+  # Print an error message and exit if the command is no longer supported
+  unsupported_cmd_message = Commands.unsupported_cmd?(cmd)
+  odie unsupported_cmd_message if unsupported_cmd_message
```

Comment:
How about `deleted_cmd` instead?

Relatedly, I'm not sure if it's worth having this level of abstraction just for `--cask`.

Reply:
> Relatedly, I'm not sure if it's worth having this level of abstraction just for `--cask`.

I think I agree. It makes sense that since cask was such an important command it can get some 
special treatment" in terms of messaging. But I don't really see many other commands getting this in the future.

Another thing to consider is that it's totally possible for someone to create a custom `brew cask` command and I'm not sure we want to totally forbid that from happening.

Instead, I'd recommend just adding an `if cmd == "cask"` around the [`odie` on line 128 of `brew.rb`](https://github.com/Homebrew/brew/blob/f18cbd2066d114ad3dec053c2df4845f4621dc6a/Library/Homebrew/brew.rb#L128) and show a custom message there instead of the existing `odie`.